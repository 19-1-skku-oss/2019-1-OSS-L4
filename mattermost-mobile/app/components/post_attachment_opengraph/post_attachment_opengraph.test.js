// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import {
    Image,
    TouchableWithoutFeedback,
    TouchableOpacity,
} from 'react-native';

import Preferences from 'mattermost-redux/constants/preferences';

import PostAttachmentOpenGraph from './post_attachment_opengraph';

describe('PostAttachmentOpenGraph', () => {
    const openGraphData = {
        site_name: 'Mattermost',
        title: 'Title',
        url: 'https://mattermost.com/',
        images: [{
            secure_url: 'https://www.mattermost.org/wp-content/uploads/2016/03/logoHorizontal_WS.png',
        }],
    };
    const baseProps = {
        actions: {
            getOpenGraphMetadata: jest.fn(),
        },
        deviceHeight: 600,
        deviceWidth: 400,
        imagesMetadata: {
            'https://www.mattermost.org/wp-content/uploads/2016/03/logoHorizontal_WS.png': {
                width: 1165,
                height: 265,
            },
        },
        isReplyPost: false,
        link: 'https://mattermost.com/',
        navigator: {},
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot, without image and description', () => {
        const wrapper = shallow(
            <PostAttachmentOpenGraph {...baseProps}/>
        );

        // should return null
        expect(wrapper.getElement()).toMatchSnapshot();

        wrapper.setProps({openGraphData});
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(TouchableOpacity).exists()).toEqual(true);
    });

    test('should match snapshot, without site_name', () => {
        const newOpenGraphData = {
            title: 'Title',
            url: 'https://mattermost.com/',
        };
        const wrapper = shallow(
            <PostAttachmentOpenGraph
                {...baseProps}
                openGraphData={newOpenGraphData}
            />
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, without title and url', () => {
        const wrapper = shallow(
            <PostAttachmentOpenGraph
                {...baseProps}
                openGraphData={{}}
            />
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match state and snapshot, on renderImage', () => {
        const wrapper = shallow(
            <PostAttachmentOpenGraph {...baseProps}/>
        );

        // should return null
        expect(wrapper.instance().renderImage()).toMatchSnapshot();
        expect(wrapper.state('hasImage')).toEqual(false);
        expect(wrapper.find(Image).exists()).toEqual(false);
        expect(wrapper.find(TouchableWithoutFeedback).exists()).toEqual(false);

        const images = [{height: 440, width: 1200, url: 'https://mattermost.com/logo.png'}];
        const openGraphDataWithImage = {...openGraphData, images};
        wrapper.setProps({openGraphData: openGraphDataWithImage});

        expect(wrapper.instance().renderImage()).toMatchSnapshot();
        expect(wrapper.state('hasImage')).toEqual(true);
        expect(wrapper.find(Image).exists()).toEqual(true);
        expect(wrapper.find(TouchableWithoutFeedback).exists()).toEqual(true);
    });

    test('should match state and snapshot, on renderDescription', () => {
        const wrapper = shallow(
            <PostAttachmentOpenGraph
                {...baseProps}
                openGraphData={openGraphData}
            />
        );

        // should return null
        expect(wrapper.instance().renderDescription()).toMatchSnapshot();

        const openGraphDataWithDescription = {...openGraphData, description: 'Description'};
        wrapper.setProps({openGraphData: openGraphDataWithDescription});
        expect(wrapper.instance().renderDescription()).toMatchSnapshot();
    });

    test('should match result on getFilename', () => {
        const wrapper = shallow(
            <PostAttachmentOpenGraph {...baseProps}/>
        );

        const testCases = [
            {link: 'https://mattermost.com/image.png', result: 'og-image.png'},
            {link: 'https://mattermost.com/image.jpg', result: 'og-image.jpg'},
            {link: 'https://mattermost.com/image', result: 'og-image.png'},
        ];

        testCases.forEach((testCase) => { // eslint-disable-line max-nested-callbacks
            expect(wrapper.instance().getFilename(testCase.link)).toEqual(testCase.result);
        });
    });
});
