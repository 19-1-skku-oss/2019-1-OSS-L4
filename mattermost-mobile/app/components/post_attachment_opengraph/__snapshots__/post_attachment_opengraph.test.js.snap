// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostAttachmentOpenGraph should match snapshot, without image and description 1`] = `null`;

exports[`PostAttachmentOpenGraph should match snapshot, without image and description 2`] = `
<View
  style={
    Object {
      "borderColor": "rgba(61,60,64,0.2)",
      "borderRadius": 3,
      "borderWidth": 1,
      "flex": 1,
      "marginTop": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "rgba(61,60,64,0.5)",
          "fontSize": 12,
          "marginBottom": 10,
        }
      }
    >
      Mattermost
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#2389d7",
              "fontSize": 14,
              "marginBottom": 10,
            },
            Object {
              "marginRight": 0,
            },
          ]
        }
      >
        Title
      </Text>
    </TouchableOpacity>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "rgba(61,60,64,0.2)",
          "borderRadius": 3,
          "borderWidth": 1,
          "marginTop": 5,
        },
        Object {
          "height": 69.83261802575107,
          "width": 307,
        },
      ]
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <Image
        resizeMode="contain"
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "height": 69.83261802575107,
              "width": 307,
            },
          ]
        }
      />
    </TouchableWithoutFeedback>
  </View>
</View>
`;

exports[`PostAttachmentOpenGraph should match snapshot, without site_name 1`] = `
<View
  style={
    Object {
      "borderColor": "rgba(61,60,64,0.2)",
      "borderRadius": 3,
      "borderWidth": 1,
      "flex": 1,
      "marginTop": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#2389d7",
              "fontSize": 14,
              "marginBottom": 10,
            },
            Object {
              "marginRight": 0,
            },
          ]
        }
      >
        Title
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`PostAttachmentOpenGraph should match snapshot, without title and url 1`] = `
<View
  style={
    Object {
      "borderColor": "rgba(61,60,64,0.2)",
      "borderRadius": 3,
      "borderWidth": 1,
      "flex": 1,
      "marginTop": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#2389d7",
              "fontSize": 14,
              "marginBottom": 10,
            },
            Object {
              "marginRight": 0,
            },
          ]
        }
      >
        https://mattermost.com/
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`PostAttachmentOpenGraph should match state and snapshot, on renderDescription 1`] = `null`;

exports[`PostAttachmentOpenGraph should match state and snapshot, on renderDescription 2`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={5}
    style={
      Object {
        "color": "rgba(61,60,64,0.7)",
        "fontSize": 13,
        "marginBottom": 10,
      }
    }
  >
    Description
  </Text>
</View>
`;

exports[`PostAttachmentOpenGraph should match state and snapshot, on renderImage 1`] = `null`;

exports[`PostAttachmentOpenGraph should match state and snapshot, on renderImage 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "rgba(61,60,64,0.2)",
        "borderRadius": 3,
        "borderWidth": 1,
        "marginTop": 5,
      },
      Object {
        "height": 112.56666666666666,
        "width": 307,
      },
    ]
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Image
      resizeMode="contain"
      style={
        Array [
          Object {
            "borderRadius": 3,
          },
          Object {
            "height": 112.56666666666666,
            "width": 307,
          },
        ]
      }
    />
  </TouchableWithoutFeedback>
</View>
`;
